mipengine/controller/controller.py

```diff
@@ -155,13 +201,517 @@ def create_data_model_views(
+            data_model=self._data_model, datasets=self._datasets
+        )
+        return [
+            self._node_landscape_aggregator.get_node_info(nodeid) for nodeid in nodeids
+        ]
+
+    def _get_globalnodeinfo(self) -> NodeInfo:
+        # TODO why is NodeLandscape raising an exception when get_global_node is called
+        # and there is no Global Node?
+        try:
+            return self._node_landscape_aggregator.get_global_node()
+        except Exception:
+            # means there is no global node, single local node execution...
+            return None
+
+    @property
+    def nodes(self) -> Nodes:
+        """
+        Returns the nodes (local nodes and global node) that have been selected based
+        on whether they contain data belonging to the "data model" and "datasets" passed
+        during the instantioation of the NodesFederation.
+        NOTE: Getting this value can be different before and after calling method
+        "create_data_model_views" if the check_min_rows flag is set to True. The reason
+        is that method "create_data_model_views" can potentially reject some nodes,
+        after applying the variables' filters and the dropna flag, on the specific
+        variable groups, since some of the local nodes might not contain "sufficient
+        data".
+        """
+        return self._nodes
+
+    @property
+    def node_ids(self):
+        local_node_ids = [node.node_id for node in self._nodes.local_nodes]
+
+        node_ids = []
+        if self._nodes.global_node:
+            node_ids = [self._nodes.global_node.node_id] + local_node_ids
+        else:
+            node_ids = local_node_ids
+
+        return node_ids
+
+    def create_data_model_views(
+        self, variable_groups: List[List[str]], dropna: bool, check_min_rows: bool
+    ) -> DataModelViews:
+        """
+        Create the appropriate view tables in the nodes databases(what is called
+        "data model views") and return a DataModelViews object
+
+        Returns
+        -------
+        DataModelViews
+            The "data model views"
+
+        """
+        data_model_views_creator = DataModelViewsCreator(
+            local_nodes=self._nodes.local_nodes,
+            variable_groups=variable_groups,
+            var_filters=self._var_filters,
+            dropna=dropna,
+            check_min_rows=check_min_rows,
+            command_id=self._command_id_generator.get_next_command_id(),
+        )
+        data_model_views_creator.create_data_model_views()
+
+        # NOTE after creating the "data model views" some of the local nodes in the
+        # original list (self._nodes.local_nodes), can be filtered out of the
+        # execution if they do not contain "suffucient data", thus
+        # self._nodes.local_nodes are updated
+        local_nodes_filtered = (
+            data_model_views_creator.data_model_views.get_list_of_nodes()
+        )
```

Comment:
what are the 3 methods you re referring to? Also, the aim is the opposite of unifying things, is to break them into smaller method/functions that do if possible only one thing in order to be unit testable

Reply:
The 3 methods that I am referring to is the `__init__`, `create_data_model_views` and the `data_model_views` property.
The only thing they are doing is creating the data model views. The init is setting variables and the data_model_views property is for getting the view, the method is not broken into pieces, it's just setters and getters with no other use.

Reply:
`__init__` is initializing the object. I see what you mean by "setter" but in general by setters we mean methods that _update_ a value, not setting it for the first time, these are initializers.
Parameters `local_nodes`,`variable_groups`,`var_filters`,`dropna`,`check_min_rows`, are attributes of the object. When instantiating a `DataModelViewsCreator` I want this `DataModelViewsCreator` to create the "data model views" specifically for the current execution, and thus for these specific parameters, so these parameters should define the internal state of the object. 
What you 're (almost) proposing is implementing what is called a utility or static class, that does not have any internal state, only static methods (`classmethods` in python). Yes this would work as well but I believe it is a matter of taste, there is no gain or loss on either approach here, I just prefer classes with internal state. 
Another minor point is that, the way I have implemented it, even if `DataModelViewsCreator.create_data_model_views` is called multiple times, by mistake, it will not create multiple views in the databases of the nodes. This is a small safeguard that would be much more difficult to implement otherwise. You can still ofcourse instantiate a new `DataModelViewsCreator` with the same parameters and create new views for the same parameters but this could potentially be a little bit more obvious mistake to spot.